From 51acec871687cdffeb907c0b9f56425841594c81 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Lo=C3=AFc=20Le=20Page?= <llepage@fluendo.com>
Date: Mon, 20 May 2019 18:56:39 +0200
Subject: [PATCH] Fix shlibs ldd call on Windows

When resolving non-libtool libraries on Windows, ldd call may receive a
path with anti-slash characters which makes ldd fail.
---
 giscanner/shlibs.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/giscanner/shlibs.py b/giscanner/shlibs.py
index e6de7bb6..288bda4a 100644
--- a/giscanner/shlibs.py
+++ b/giscanner/shlibs.py
@@ -100,7 +100,7 @@ def _resolve_non_libtool(options, binary, libraries):
         if platform_system == 'Darwin':
             args.extend(['otool', '-L', binary.args[0]])
         else:
-            args.extend(['ldd', binary.args[0]])
+            args.extend(['ldd', binary.args[0].replace('\\', '/')])
         output = subprocess.check_output(args)
         if isinstance(output, bytes):
             output = output.decode("utf-8", "replace")
-- 
2.21.0.windows.1

